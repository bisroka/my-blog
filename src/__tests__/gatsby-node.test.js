const { createPages } = require('../../gatsby-node')
const path = require('path')

jest.mock('path', () => ({
  resolve: jest.fn(),
}))



describe('testing gatsby-node', () => {
    describe('given data', () => {
    const data = {
        allDatoCmsArticle: {
            edges: {
            title: 'SZÓSTY WPIS PRZYKŁADOWO TESTOWY',
            slug: 'szosty-wpis-przykladowo-testowy',
            content: 'szósty wpis przykładowo testowy',
            thumbnail: {
                url: 'https://www.datocms-assets.com/31772/1595523423-zrzut-ekranu-2020-07-15-o-23-57-53.png'
            }, 
            }
      }
    }

    describe('when querying for edges', () => {
      const graphql = jest.fn(() =>
        Promise.resolve({
          data,
        })
      )
      const createPage = jest.fn()
      const ChapterComponent = () => {}

      path.resolve.mockImplementation(() => ChapterComponent)

      beforeEach(() => {
        createPages({ graphql, actions: { createPage } })
      })

      it('then runs a graphql query', () => {
        expect(graphql.mock.calls[0][0]).toMatchSnapshot()
      })

      // it('then creates a page for each chapter', () => {
      //   const article = {
      //     title: 'SZÓSTY WPIS PRZYKŁADOWO TESTOWY',
      //     slug: 'szosty-wpis-przykladowo-testowy',
      //     content: 'szósty wpis przykładowo testowy',
      //     thumbnail: {
      //         url: 'https://www.datocms-assets.com/31772/1595523423-zrzut-ekranu-2020-07-15-o-23-57-53.png'
      //     }, 
      //   }
      //  expect(createPage).toHaveReturned({
      //       path: `blog/${article.slug}`,
      //       context: article.slug,
      //     })
      //   })
      // })
    })
    })
  })