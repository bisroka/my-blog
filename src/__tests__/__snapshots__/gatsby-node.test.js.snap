// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing gatsby-node given data when querying for chapters then runs a graphql query 1`] = `
"
    {
      allDatoCmsArticle {
        edges {
          node {
            title
            slug
            content
            thumbnail {
              url
            }
          }
        }
      }
    }
  "
`;

exports[`testing gatsby-node given data when querying for edges then runs a graphql query 1`] = `
"
    {
      allDatoCmsArticle {
        edges {
          node {
            title
            slug
            content
            thumbnail {
              url
            }
          }
        }
      }
    }
  "
`;
